.github/workflows/main.yml

```diff
@@ -1,30 +1,41 @@
 name: Screenshots
 
-# Run on any commit or pr on any branch
+# Run on any pr on any branch
 on:
   pull_request:
     branches:
       - '**'
 
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
-  build:
+  screenshots:
     runs-on: ubuntu-latest
-    
     steps:
+    - name: Check whether unity activation requests should be done
+      id: checksecret
+      env:
+          KEY: ${{ secrets.BROWSERSTACK_KEY }}
+      run: |
+          echo "is_BROWSERSTACK_KEY_set: ${{ env.KEY != '' }}"
+          echo "::set-output name=is_BROWSERSTACK_KEY_set::${{ env.KEY != '' }}"
+          
     # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
     - uses: actions/checkout@v2
+      #Checks if the browserstack key exists
+      if: ${{ steps.checksecret.outputs.is_BROWSERSTACK_KEY_set == 'true' }}
 
     - name: Generate Website Screenshots
+    #Checks if the browserstack key exists
+      if: ${{ steps.checksecret.outputs.is_BROWSERSTACK_KEY_set == 'true' }}
       run: |
          pip3 install --upgrade pip
          export BROWSERSTACK_KEY=${{secrets.BROWSERSTACK_KEY}}
          pip3 install selenium
```

Comment:
this is daniel's doing but why aren't we using `requirements.txt`

Reply:
What's `requirements.txt`?

Reply:
from: https://pip.pypa.io/en/stable/user_guide/
![image](https://user-images.githubusercontent.com/24723940/117084250-94506e80-acfb-11eb-897a-3816a49fb93b.png)

like i see the file that has the requirements and we're sorta hard coding selenium in. I don't think it matters since it's probably always going to be just selenium. I'm just not sure why we have requirements.txt then @daniel-sudz

Reply:
![image](https://user-images.githubusercontent.com/24723940/117084318-c235b300-acfb-11eb-9711-f52014af7151.png)

Reply:
It's just
`selenium==3.141.0`

Reply:
Nothing to do with your PR so you can resolve this haha sorry

Reply:
@daniel-sudz

Reply:
Not sure what the question is but ```requirments.txt``` is similar to ```package.json``` in functionality